hathor/transaction/token_info.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import IntEnum
+
+
+class TokenInfoVersion(IntEnum):
+    DEPOSIT = 1
+    FEE = 2
+
+
+# used when (de)serializing token information
+@dataclass(slots=True, kw_only=True)
+class TokenInfo:
+    amount: int
+    can_mint: bool
+    can_melt: bool
+    version: TokenInfoVersion | None
+
+    @classmethod
+    def get_default(cls, version: TokenInfoVersion | None = TokenInfoVersion.DEPOSIT,
+                    is_new_token: bool = False) -> TokenInfo:
```

Comment:
What does `is_new_token` mean? Why is it used in the `can_mint` and `can_melt` attributes? I'd rather have two parameters (`can_mint` and `can_melt`) and let the caller decide individually on them.